### PR TITLE
Fix: If FIO test running in background is stuck throw the exception and abort the test

### PR DESCRIPTION
**Pull request by prashantmhetar**
Bug #367 

Below code changes handles the ./StartFioTest.sh stuck issue with checking the status string and abort the test with throwing the exception.
To verify this fix running performance tests to reproduce the scenario

### CODE REVIEW

File: Testscripts/Windows/PERF-STORAGE-MULTIDISK-RAID0-FIO.ps1 [hunk 1]
```diff
@@ -234,11 +234,19 @@ function Main {
...
+        $MaxCount = 10
         while ((Get-Job -Id $testJob).State -eq "Running") {
             $currentStatus = Run-LinuxCmd -ip $allVMData.PublicIP -port $allVMData.SSHPort `
                 -username "root" -password $password -command "tail -1 fioConsoleLogs.txt" `
                 -runAsSudo
             Write-LogInfo "Current Test Status: $currentStatus"
+            if ($currentStatus -imatch "Doing forceful exit of this job") {
+                $ForceExitCount++
+                if ( $ForceExitCount -eq $MaxCount) {
+                    throw "FIO is stuck, exiting the test"
```
Comment (iamshital) [inline Testscripts/Windows/PERF-STORAGE-MULTIDISK-RAID0-FIO.ps1]: let's change 'exiting the test' -> 'aborting the test.'

File: Testscripts/Windows/PERF-STORAGE-MULTIDISK-RAID0-FIO.ps1 [hunk 2]
```diff
@@ -234,11 +234,19 @@ function Main {
             -RunInBackground -runAsSudo
 
         Write-LogInfo "Monitoring test run..."
+        $ForceExitCount = 0
```
Comment (iamshital) [inline Testscripts/Windows/PERF-STORAGE-MULTIDISK-RAID0-FIO.ps1]: This value needs to reset in While loop, if the $current status changes from "Doing forceful exit of this job" --> "next iteration" . And change the variable name to - `$FioStuckCounter`

File: Testscripts/Windows/PERF-STORAGE-MULTIDISK-RAID0-FIO.ps1 [hunk 3]
```diff
@@ -234,11 +234,19 @@ function Main {
...
+        $MaxCount = 10
         while ((Get-Job -Id $testJob).State -eq "Running") {
             $currentStatus = Run-LinuxCmd -ip $allVMData.PublicIP -port $allVMData.SSHPort `
```
Comment (iamshital) [inline Testscripts/Windows/PERF-STORAGE-MULTIDISK-RAID0-FIO.ps1:240]: Add this code here : 
```
if ($currentStatus -inotmatch "Doing forceful exit of this job") {
    $FioStuckCounter = 0
}
```

File: Testscripts/Windows/PERF-STORAGE-MULTIDISK-RAID0-FIO.ps1 [hunk 4]
```diff
@@ -234,11 +234,19 @@ function Main {
...
                 -username "root" -password $password -command "tail -1 fioConsoleLogs.txt" `
                 -runAsSudo
             Write-LogInfo "Current Test Status: $currentStatus"
+            if ($currentStatus -imatch "Doing forceful exit of this job") {
+                $ForceExitCount++
+                if ( $ForceExitCount -eq $MaxCount) {
```
Comment (iamshital) [inline Testscripts/Windows/PERF-STORAGE-MULTIDISK-RAID0-FIO.ps1]: $MaxCount --> $MaxFioStuckAttempts

File: Testscripts/Windows/PERF-STORAGE-OVER-NFS.ps1 [hunk 5]
```diff
@@ -84,9 +84,17 @@ chmod 666 /root/perf_fio.csv
         #endregion
 
         #region MONITOR TEST
```
Comment (iamshital) [inline Testscripts/Windows/PERF-STORAGE-OVER-NFS.ps1:86]: Same changes as per above file.

**Review by iamshital — CHANGES_REQUESTED** [review]

File: Testscripts/Windows/PERF-STORAGE-MULTIDISK-RAID0-FIO.ps1 (hunk 1 above)
Comment (prashantmhetar) [reply Testscripts/Windows/PERF-STORAGE-MULTIDISK-RAID0-FIO.ps1]: modified exception log

File: Testscripts/Windows/PERF-STORAGE-MULTIDISK-RAID0-FIO.ps1 (hunk 2 above)
Comment (prashantmhetar) [reply Testscripts/Windows/PERF-STORAGE-MULTIDISK-RAID0-FIO.ps1]: done

File: Testscripts/Windows/PERF-STORAGE-MULTIDISK-RAID0-FIO.ps1 (hunk 3 above)
Comment (prashantmhetar) [reply Testscripts/Windows/PERF-STORAGE-MULTIDISK-RAID0-FIO.ps1:240]: added else condition to reset the counter

File: Testscripts/Windows/PERF-STORAGE-MULTIDISK-RAID0-FIO.ps1 (hunk 4 above)
Comment (prashantmhetar) [reply Testscripts/Windows/PERF-STORAGE-MULTIDISK-RAID0-FIO.ps1]: renamed the variable

File: Testscripts/Windows/PERF-STORAGE-OVER-NFS.ps1 (hunk 5 above)
Comment (prashantmhetar) [reply Testscripts/Windows/PERF-STORAGE-OVER-NFS.ps1:86]: above same changes are incorporated in this file

**Review by LiliDeng — APPROVED** [review]
Could you please post results against latest code?

**Comment by prashantmhetar** [comment]
> Could you please post results against latest code?

Tried to reproduce this issue with pipeline-cloudtest-manual jobs 1647,1648, 1696, 1697

**Comment by prashantmhetar** [comment]
> > Could you please post results against latest code?
> 
> Tried to reproduce this issue with pipeline-cloudtest-manual jobs 1647,1648, 1696, 1697. Please refer the jobs for latest results

**Review by iamshital — APPROVED** [review]

**Pull request merged**